libs/elasticsearch-core/src/main/java/org/elasticsearch/common/collect/Tuple.java

```diff
@@ -46,6 +46,7 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
 
+        @SuppressWarnings("rawtypes")
```

Comment:
The warning can be avoided by doing an instanceof check, eg:

```
if (o instanceof Tuple<?, ?>) {
  // rest of current code
}
return false;
```

Reply:
That doesn't prevent the warning/error, I had to change it to

```java
        Tuple<?, ?> tuple = (Tuple<?, ?>) o;
```

Reply:
Good to know!